### ISSUE TITLE
Dungeon Defenders 2 (236110)

### ISSUE THREAD

**Issue by tomilepp** [ValveSoftware/Proton#393]
Played couple of hours. All seems to work.
Fedora 28 / Nvidia GTX 1060

[system info](https://gist.github.com/tomilepp/454b25340515d5f5607dfbeb8ee4d5fb)

**Comment by kisak-valve** [comment]
Dungeon Defenders II (236110)

Issue transferred from https://github.com/ValveSoftware/Proton/issues/1826.
@ProfessorChill posted on 2018-10-27T16:47:31:

# Whitelist Request
- Name of the game to be whitelisted: Dungeon Defenders II
- Steam AppID of the game: 236110

## System Information
- GPU: GTX 980
- Driver/LLVM version: nvidia 410.66-4
- Kernel version: 4.19.arch1-1
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/ProfessorChill/761ec5bbd91ea558daec39db4b216442
- Proton version: 3.7-8

## I confirm:
- [X] that pressing the `Play` button in the Steam client is sufficient.
- [ ] that [runtime config options](https://github.com/ValveSoftware/Proton#runtime-config-options) are necessary to run the game.
- [x] that no workarounds other than the mentioned ones are necessary.

## Issues
- [X] I haven't experienced any issues.
- [X] There are no issues left open for this game.
- [ ] Although I consider the gaming experience equal to Windows there are remaining issues:

This is not an issue so I am not going to check it as an issue.
The game is laggy as heck, this could impact gameplay a bit.

**Comment by jomarocas** [comment]
i confimr the game works good with steam controller and game works good, except the framerate is variable

**Comment by DataBeaver** [comment]
I disagree about the game working well.  It does start up and the town and tavern look good, but actual in-game graphics are very buggy.

![screenshot](https://steamuserimages-a.akamaihd.net/ugc/952977739277922296/76C2E4A0F487001E86CAAD9A1181A1C39310E2A1/)

- most of the ground is black (probably buggy lighting or shadow map)
- minimap is missing the actual map and only shows various markers
- enemies are shown as sort of ghosts with only outlines visible

I was able to complete the map but that was largely thanks to having played it before (so I knew where to build) and my heroes being overpowered for this difficulty (so it didn't matter if I couldn't play very efficiently due to the bad graphics).  When I selected "to tavern" from the end-of-map summary screen, nothing happened for a while and then the game informed me that it couldn't find a session for me.  Then it exited to desktop.  I am uncertain if this is an issue with the game or with Proton.

For the most part the game runs at 60 fps, though there are intermittent lag spikes (possibly due to loading something or other over NFS).  Those would be a relatively minor issue if the graphics were correct.

GPU: RTX 2070
Driver: Nvidia 410.93
Kernel: 4.18.11 (custom build)
Full system report: https://gist.github.com/DataBeaver/d032d93469df5b5d1e6189756ca161fc (The part that says laptop is wrong; I have no idea where Steam is pulling that from)
Proton version: Proton 3.16-6 beta

**Comment by kisak-valve** [comment]
Hello @DataBeaver, it may be worthwhile to test a newer series of nvidia driver.

**Comment by DataBeaver** [comment]
Looks like this doesn't happen every time, and not always on the same maps.  Upon restarting the game I was able to play two maps (including the one that bugged out previously) without issues.  On the third one the graphics were again bad and lag spikes appeared.  I was also again unable to return to the tavern, so I wonder if these are somehow connected.

My current drivers are the latest available on Debian experimental, but I'm no stranger to packaging newer versions so I will try that.

I'll also try moving the game to a local disk.  It might be that in certain situations NFS causes resource loads to time out.

**Comment by DataBeaver** [comment]
Moving the game to a local disk did not help.  The graphics issues still crop up after three or four maps.  I'm thinking this might be a resource exhaustion problem due to a 32-bit binary.  I had vaguely similar problems with Final Fantasy XIV (non-Steam version) and they were eventually solved when I got the 64-bit DX11 version to work.  Unfortunately there is no 64-bit version of Dungeon Defenders 2.

On my initial attempt after installing the game I got the issue on the very first map, probably because the game put me in the town by default after login.  There's a lot of different players with different gear there, which causes many different resources to be loaded.

I will still try the more recent drivers, though if my suspicion of resource exhaustion issue is correct, it will not help.

**Comment by DataBeaver** [comment]
@kisak-valve I updated to the brand spanking new 418.30 beta drivers from Nvidia and the graphics bug remains unchanged.

**Comment by DataBeaver** [comment]
Repeating the same map without visiting the tavern in between works fine.  I've done over ten repetitions in a row without encountering issues.  Likely the game reuses already loaded resources for the map, so this observation lends credence to the theory that the issue is resource exhaustion.

**Comment by BloodyIron** [comment]
I get VERY poor performance in Dungeon Defenders 2 with STEAM Play

**Comment by kisak-valve** [comment]
Dungeon Defenders 2

Issue transferred from https://github.com/ValveSoftware/Proton/issues/3883.
@Gibbz posted on 2020-05-16T23:42:29:

# Compatibility Report
- Name of the game with compatibility issues: Dungeon Defenders II
- Steam AppID of the game:
236110
https://store.steampowered.com/app/236110/Dungeon_Defenders_II/

## System Information
- GPU: RX 580
- Driver/LLVM version: Mesa 20.0.6
- Kernel version: 5.6
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/Gibbz/9599aca1fa5485d1584e4837b72bd976
- Proton version: 5.0-7

## I confirm:
- [ x] that I haven't found an existing compatibility report for this game.
- [ x] that I have checked whether there are updates for my system available.



## Symptoms 
The game crashes/freezes after some amount of time. Usually seems to be every 2-3 games it will suddenly freeze requiring the user to kill the game

## Reproduction
It seems very intermittent, but have some logs.
[steam-236110_2.log](https://github.com/ValveSoftware/Proton/files/4639417/steam-236110_2.log)
[steam-236110.log](https://github.com/ValveSoftware/Proton/files/4639419/steam-236110.log)

**Comment by BloodyIron** [comment]
I'm actually getting really good performance with Proton for Dungeon Defenders 2 now, yay!

However, I have audio stuttering, not sure what to do...
